app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -577,6 +577,32 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
     bool ringPinching = (mAimState.status & XR_HAND_TRACKING_AIM_RING_PINCHING_BIT_FB) != 0;
     delegate.SetButtonState(mIndex, ControllerDelegate::BUTTON_APP, -1, ringPinching,
                             ringPinching, 1.0);
+
+    // Prepare and submit hand joint locations data for rendering
+    assert(mHasHandJoints);
+    std::vector<vrb::Matrix> jointTransforms;
+    jointTransforms.resize(mHandJoints.size());
+    for (int i = 0; i < mHandJoints.size(); i++) {
+        vrb::Matrix transform = XrPoseToMatrix(mHandJoints[i].pose);
+        if (mHandJoints[i].locationFlags & XR_SPACE_LOCATION_POSITION_VALID_BIT) {
+            vrb::Vector correctedPosition{
```

Comment:
This deserves a comment, since the code does not explain why we need this for.

Reply:
ok, will add a comment.

Reply:
Added a comment, and also a guard for the correction to apply only to OculusVR, pending to see how's the story for Pico. The correction should not be necessary as I explained in the code comment.